calico/openstack/t_etcd.py

```diff
@@ -235,6 +256,27 @@ def provide_felix_config(self):
             self.client.write(READY_KEY, 'true')
 
     @_handling_etcd_exceptions
+    def get_endpoint_data(self, endpoint):
+        """
+        Get data for an endpoint out of etcd. This should be used on endpoints
+        returned from functions like ``get_endpoints``.
+
+        :param endpoint: An ``Endpoint`` class.
+        :return: A ``Endpoint`` class with ``data`` not None.
+        """
+        LOG.debug("Getting endpoint %s", endpoint.id)
+
+        result = self.client.read(endpoint.key, timeout=5)
```

Comment:
We're using these timeouts in a few places, worth wrapping up so we don't forget them and only have one place to change  at 3am when the timeout is too short.